trieste/ask_tell_optimization.py

```diff
@@ -38,6 +39,9 @@
 SP = TypeVar("SP", bound=SearchSpace)
 """ Type variable bound to :class:`SearchSpace`. """
 
+M_contra = TypeVar("M_contra", bound=TrainableProbabilisticModel, contravariant=True)
```

Comment:
huh, no clue what is this for... can you recommend a good source to learn about these contravariants etc?